src/business-logic/services/test/device-service.spec.ts

```diff
@@ -0,0 +1,41 @@
+import { DeviceServiceImplementation } from '../device-service-implementation'
+import { Device, TelemetricsDevice } from '../../../model/entities/device'
+import { mock, when, verify, instance } from '@typestrong/ts-mockito'
+import { MongoDeviceRepository } from '../../../data-access/repositories/mongo/mongo-device-repository'
+import { DeviceService } from '../interfaces/device-service'
+import { DeviceEventEmitter } from '../interfaces/device-event-emitter'
+
+describe('DeviceService', () => {
+  let testDevice: TelemetricsDevice
+  let deviceRepository: MongoDeviceRepository
+  let deviceEventEmitter: DeviceEventEmitter
+  const mockedValue = mock<MongoDeviceRepository>() 
```

Comment:
This is "global" mock setup. Refer to my other comment about "global" mock setup in this file.